frontend/packages/kubevirt-plugin/src/utils/validations/template/template-validations.ts

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable lines-between-class-members */
+import * as _ from 'lodash';
+import { humanizeBinaryBytes } from '@console/internal/components/utils';
+import { ValueEnum } from '../../../constants';
+import { CommonTemplatesValidation } from '../../../types/template';
+import { IntegerValidationResult } from './types';
+
+// TODO: Add all the fields in the form
+export class ValidationJSONPath extends ValueEnum<string> {
+  static readonly CPU = new ValidationJSONPath('jsonpath::.spec.domain.cpu.cores');
+  static readonly MEMORY = new ValidationJSONPath(
+    'jsonpath::.spec.domain.resources.requests.memory',
+  );
+}
+
+export class TemplateValidations {
+  private validations: CommonTemplatesValidation[];
+
+  constructor(validations: CommonTemplatesValidation[] = []) {
+    this.validations = _.compact(validations);
+  }
+
+  validateMemory = (value: number): IntegerValidationResult => {
+    const { min, max, isMinInclusive, isMaxInclusive, isValid } = this.validateInteger(
+      value,
+      ValidationJSONPath.MEMORY,
+      {
+        defaultMin: 0,
+        isDefaultMinInclusive: false,
+      },
+    );
+
+    let errorMsg = null;
+
+    if (!isValid) {
+      if (min !== 0 && Number.isFinite(min) && Number.isFinite(max)) {
+        errorMsg = `Memory must be between ${humanizeBinaryBytes(min).string} and ${
+          humanizeBinaryBytes(max).string
+        }`;
+      } else if (Number.isFinite(max)) {
+        errorMsg = `Memory must be below ${humanizeBinaryBytes(max).string}`;
+      } else {
+        errorMsg = `Memory must be above ${humanizeBinaryBytes(min).string}`;
+      }
+    }
+    return { isValid, errorMsg, min, max, isMinInclusive, isMaxInclusive };
+  };
+
+  private validateInteger = (
+    value: number,
+    jsonPath: ValidationJSONPath,
+    {
+      isDefaultMinInclusive = true,
+      isDefaultMaxInclusive = true,
+      defaultMin = Number.NEGATIVE_INFINITY,
+      defaultMax = Number.POSITIVE_INFINITY,
+    },
+  ) => {
+    const relevantValidations = this.getRelevantValidations(jsonPath);
+
+    // combine validations for single template and make them strict (all integer validations must pass)
+    const { min, max, isMinInclusive, isMaxInclusive } = relevantValidations.reduce(
```

Comment:
We should consider the option that a validation is optional - a.k.a has the key `justWarning`.
optional validation shouldn't be combined and enforced.

Reply:
right! added the support for `justWarning`